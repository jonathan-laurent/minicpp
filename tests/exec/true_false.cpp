#include <iostream>

int main() {
  int b = true;
  std::cout << "b = " << b << "\n";
  b = false;
  std::cout << "b = " << b << "\n";
}
