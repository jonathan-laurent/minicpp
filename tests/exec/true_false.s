	.text
	jal  _p0__main
	li   $v0, 10
	syscall
_p0__main:
	addi $sp, $sp, -4
	li   $t0, 1
	la   $a0, _s__1
	li   $v0, 4
	syscall
	move $a0, $t0
	li   $v0, 1
	syscall
	la   $a0, _s__0
	li   $v0, 4
	syscall
	li   $t0, 0
	la   $a0, _s__3
	li   $v0, 4
	syscall
	move $a0, $t0
	li   $v0, 1
	syscall
	la   $a0, _s__2
	li   $v0, 4
	syscall
	addi $sp, $sp, 4
	jr   $ra
	.data
_s__0:
	.asciiz "\n"
_s__1:
	.asciiz "b = "
_s__2:
	.asciiz "\n"
_s__3:
	.asciiz "b = "

