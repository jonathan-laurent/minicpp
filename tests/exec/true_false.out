b = 1
b = 0
